Add MCP Server config view to AI Configuration

sgraband opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See SECURITY.md at the root of this repository, to learn how to report
vulnerabilities.
-->

#### What it does

<!-- Include relevant issues and describe how they are addressed. -->
[Improve MCP services](https://github.com/eclipse-theia/theia/commit/cd6b55ba5775c8317559c2f20d3e64e3855f1565) 

Introduce notification mechanism to listen to MCP server changes.
Created common interface for MCPFrontendService.
Rename getStartedServers() to getRunningServers().
Added status handling to MCPServer (NotRunning, Starting, Running, Errored):
Add getServerDescription endpoint to retrieve information about the server.
Add status, error message and tool information to MCPServerDescription.

[Add MCP Config view to AI Configuration](https://github.com/eclipse-theia/theia/commit/d94c53acc7cad15e1efe20dfda0eb3879df54b85) 

This view helps to see the state of MCP servers.
Displays all the created settings. Obfuscates tokens (very basic support).
Shows the status of a server (Running, Starting, Errored, Not Running).
Shows all tools available. Let's the user easily copy the tools for chat and prompttemplate use.
Offers a button to start/stop the view.

[Fix autostart stopping for following servers if one fails](https://github.com/eclipse-theia/theia/commit/82651a6d4d24d35545ade80bd6ab68fdc193fad3) 

Before this the autostart would stop for all following servers if one failed.
This was due to uncatched errors when trying to resolve the tools.
To prevent this in the future, improved the handling of the getTools() on client side.
Also adapted the Start Command to use the new status reporting instead of calling getTools() again.
#### How to test
Check out the code and play around with the view and MCP servers.
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

#### Follow-ups

<!-- Please list potential follow-up work, including known issues, possible future work, identified technical debt, and potentially introduced technical debt. If the PR introduces technical debt, specify the reason why this is acceptable. Please create tickets and link them here. Please use the label "technical debt" for new issues when it applies. -->
https://github.com/eclipse-theia/theia/issues/15306
#### Breaking changes

- [ ] This PR introduces breaking changes and requires careful review. If yes, the breaking changes section in the [changelog](https://github.com/eclipse-theia/theia/blob/master/CHANGELOG.md) has been updated.

#### Attribution

<!-- If the changelog entry for this change should contain an attribution at the end (e.g. Contributed on behalf of x) add it in this section -->

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

eneufeld commented on packages/ai-ide/src/browser/ai-configuration/mcp-configuration-widget.tsx:
```diff
@@ -0,0 +1,328 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
```
```suggestion
// Copyright (C) 2025 EclipseSource GmbH.
```

eneufeld commented on packages/ai-ide/src/browser/ai-configuration/mcp-configuration-widget.tsx:
```diff
@@ -0,0 +1,328 @@
+    static readonly ID = 'ai-mcp-configuration-container-widget';
+    static readonly LABEL = 'MCP Servers';
+
+    servers: MCPServerDescription[] = [];
```
does this needs to be public?

eneufeld commented on packages/ai-ide/src/browser/ai-configuration/mcp-configuration-widget.tsx:
```diff
@@ -0,0 +1,328 @@
+
+    protected async loadServers(): Promise<void> {
+        const serverNames = await this.mcpFrontendService.getServerNames();
+        const promises = serverNames.map(name => this.mcpFrontendService.getServerDescription(name));
```
promises is very generic, maybe use something like: descriptionPromises or just merge this and the next line?

eneufeld reviewed (changes requested):
Looks good. This is a nice addition.
- For me the black font on the green background is hard to read (Theia in light mode). In dark mode white font on green is fine.
- I don't get the empty box icon in the `Stop Server` button.
- It would be great to be able to also copy the new promptTemplate Jonas introduced.
- some nitpicks inline.
As a follow up: 
- enable/disable autostart from the view, 
- add new servers, see https://github.com/anaisbetts/mcp-installer

sgraband commented:
@eneufeld I addressed your feedback comments and added the possibility to also copy the one-liner prompttemplate string.

> For me the black font on the green background is hard to read (Theia in light mode). In dark mode white font on green is fine.

I see what you mean, but imho i am using the correct variables and they are not really suitable together in some themes, isn't this a theme issue then? I can also change the text to white all the time or use the description foreground, but it feels like working around an issue in the theming. WDYT?

> enable/disable autostart from the view,
add new servers, see https://github.com/anaisbetts/mcp-installer

I agree that these are good follow ups. I will create a ticket once this PR is approved.

eneufeld commented:
For the colors let's raise an issue or add it to the follow up.
